### PR TITLE
Add tutorial on removing the address field from person

### PR DESCRIPTION
**Pull request by j-lum**

### CODE REVIEW

File: docs/tutorial/RemovingFields.adoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+:source-highlighter: highlightjs
+endif::[]
+
+[quote, Charles Maturin, Motto to "Bertram"]
+'Tis well to be merry and wise,
+'Tis well to be honest and true;
+'Tis well to be off with the old love,
+Before you are on with the new.
+
+{empty} +
```
Comment (damithc) [inline docs/tutorial/RemovingFields.adoc:21]: Let's stick to SE quotes only :-)

File: docs/tutorial/RemovingFields.adoc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+== Introduction
+When working on Address Book, you will most likely find that some features and fields that are no longer necessary.
+Consider refactoring the existing `Person` model to suit your use case.
```
Comment (damithc) [inline docs/tutorial/RemovingFields.adoc:25]: Removing a field may not be strictly _refactoring_

File: docs/tutorial/RemovingFields.adoc [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+In this tutorial, we'll do exactly just that and remove the `Address` field from `Person`.
+
+== Safely deleting `Address`
+
+Fortunately, the IntelliJ IDEA provides a robust refactoring tool that can identify _most_ usages. Let's try to use
+it as much as we can.
+
+=== Assisted refactoring
+We start by identifying ``Address``'s usages. Select the `Address` class and perfom a `Refactor` >
+`Safe Delete` through the menu.
```
Comment (damithc) [inline docs/tutorial/RemovingFields.adoc:36]: >Select the `Address` class

Select where?

File: docs/tutorial/RemovingFields.adoc [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+. Select `Yes` when prompted to remove getters and setters
+. Select `View Usages` again image:UnsafeDeleteOnField.png[width=1145px, height=583px]
+. Remove the usages of `address` and select `Do refactor` when you are done.
+. Repeat the steps for the remaining usages of `Address`
+
+After you are done, verify that the application still works by compiling and running it again.
+
+=== Manual refactoring
+
+Unfortunately, there are usages of `Address` that IntelliJ IDEA cannot identify. Let's fix them.
```
Comment (damithc) [inline docs/tutorial/RemovingFields.adoc:61]: fix -> remove?

File: docs/tutorial/RemovingFields.adoc [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
```
Comment (damithc) [inline docs/tutorial/RemovingFields.adoc:62]: Would a project-wide text search help here? i.e., Ctrl+Shift+F

File: docs/tutorial/RemovingFields.adoc [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+After removing the `Label`, we can proceed to formally test our code. If everything went well, you should have most
+of your tests pass. Fix any remaining errors until the tests all pass.
+
+== Tidying up
+
+At this point, your application is working as intended and all your tests are passing. What's left to do is to clean
+up test data and references to `Address` in the documentation.
+
+A simple way to generate valid test data is to delete the storage file `data/addressbook.json` and use your
+application to add dummy values. Don't forget to modify a copy of valid data to produce invalid data!
```
Comment (damithc) [inline docs/tutorial/RemovingFields.adoc:96]: The reader might not be able to understand this paragraph (i.e., it assumes some background knowledge the reader might not have). Perhaps we should have some description of how testing and test data works somewhere?

File: docs/tutorial/RemovingFields.adoc [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+...
+----
+
+After removing the `Label`, we can proceed to formally test our code. If everything went well, you should have most
+of your tests pass. Fix any remaining errors until the tests all pass.
+
+== Tidying up
+
+At this point, your application is working as intended and all your tests are passing. What's left to do is to clean
+up test data and references to `Address` in the documentation.
```
Comment (damithc) [inline docs/tutorial/RemovingFields.adoc:93]: reader: If the tests are passing, why there is a need to clean up further? 🤔

**Review by damithc — CHANGES_REQUESTED** [review]
Looks good 👍 
Some suggestions added.

**Comment by j-lum** [comment]
Closing this in favor of the newer PRs.

**Pull request closed without merging**